src/clims/services/container.py

```diff
@@ -52,6 +52,10 @@ def from_any_type(cls, container, key):
             return cls.from_string(container, key)
         elif isinstance(key, tuple):
             return cls(container, *key)
+        elif issubclass(type(key), ContainerIndex):
+            # This has to be refactored in case there is problem
+            assert container == key.container, "containers are not matching in container index"
```

Comment:
Good catch. Another option here is just to ignore the container lying on the index and act as it's not there. A starting point would be to make it private-looking so that it will definitely only be used by underlying code. But for now I think this assert is good so it reminds us we need to finalize this design.